### ISSUE TITLE
01-logistic-regression.ipynb

### ISSUE THREAD

**Issue by mezcolantriz** [4GeeksAcademy/Antonio27M-machine-learning#2]
@Antonio27M 

### **Error: Scaling incorrecto en test**

**Problema principal:**

Antonio usas `fit_transform()` en el conjunto de test, causando **data leakage severo**:

<img width="1085" height="195" alt="Image" src="https://github.com/user-attachments/assets/a4dde3d4-01ef-4c6a-bd8e-1148543ba283" />

**Consecuencias:**
- El scaler calcula media y desviación con los datos de **test**
- El modelo "ve" información del test durante el entrenamiento
- Las métricas reportadas (90.36% accuracy) están infladas y no son confiables

**Corrección:**
```python
X_test_num = pd.DataFrame(norm_scaler.transform(X_test_encoding[num_variables]),
                                     ^^^^^^^^^
                          index=X_test_encoding.index, columns=num_variables)
```

---

#### 1. **`drop_duplicates()` sin `inplace=True`**
```python
#  No hace nada
df.drop_duplicates()  # El resultado no se guarda
```

Debería ser:
```python
df = df.drop_duplicates()
# o
df.drop_duplicates(inplace=True)
```

**Fortalezas:**

- ✅ Split antes de encoding
- ✅ OneHotEncoder con fit/transform correcto
- ✅ Uso de OneHotEncoder para variables nominales
- ✅ Análisis de multicolinealidad
- ✅ Concatenación correcta de variables
- ✅ Guardas el encoder con pickle
